tests/real/enterprise.rs

```diff
@@ -0,0 +1,321 @@
+        .build();
+
+    // enquere and then fetch job2, but after ttl:
+    producer.enqueue(job2).unwrap();
+    thread::sleep(time::Duration::from_secs(job_ttl_secs * 2));
+    let had_job = consumer.run_one(0, &["default"]).unwrap();
+
+    // For the non-enterprise edition of Faktory, this assertion will
+    // fail, which should be take into account when running the test suite on CI.
+    assert!(!had_job);
+}
+
+#[test]
+fn ent_unique_job() {
+    use faktory::error;
+    use serde_json::Value;
+
+    skip_if_not_enterprise!();
+
+    let url = learn_faktory_url();
+
+    let job_type = "order";
+
+    // prepare producer and consumer:
+    let mut producer = Producer::connect(Some(&url)).unwrap();
+    let mut consumer = ConsumerBuilder::default();
+    consumer.register(job_type, |job| -> io::Result<_> {
+        Ok(eprintln!("{:?}", job))
+    });
+    let mut consumer = consumer.connect(Some(&url)).unwrap();
+
+    // Reminder. Jobs are considered unique for kind + args + queue.
+    // So the following two jobs, will be accepted by Faktory, since we
+    // are not setting 'unique_for' when creating those jobs:
+    let queue_name = "ent_unique_job";
+    let args = vec![Value::from("ISBN-13:9781718501850"), Value::from(100)];
+    let job1 = JobBuilder::new(job_type)
+        .args(args.clone())
+        .queue(queue_name)
+        .build();
+    producer.enqueue(job1).unwrap();
+    let job2 = JobBuilder::new(job_type)
+        .args(args.clone())
+        .queue(queue_name)
+        .build();
+    producer.enqueue(job2).unwrap();
+
+    let had_job = consumer.run_one(0, &[queue_name]).unwrap();
+    assert!(had_job);
+    let had_another_one = consumer.run_one(0, &[queue_name]).unwrap();
+    assert!(had_another_one);
+    let and_that_is_it_for_now = !consumer.run_one(0, &[queue_name]).unwrap();
+    assert!(and_that_is_it_for_now);
+
+    // let's now create a unique job and followed by a job with
+    // the same args and kind (jobtype in Faktory terms) and pushed
+    // to the same queue:
+    let unique_for_secs = 3;
+    let job1 = Job::builder(job_type)
+        .args(args.clone())
+        .queue(queue_name)
+        .unique_for(unique_for_secs)
+        .build();
+    producer.enqueue(job1).unwrap();
+    // this one is a 'duplicate' ...
+    let job2 = Job::builder(job_type)
+        .args(args.clone())
+        .queue(queue_name)
+        .unique_for(unique_for_secs)
+        .build();
+    // ... so the server will respond accordingly:
+    let res = producer.enqueue(job2).unwrap_err();
+    if let error::Error::Protocol(error::Protocol::Internal { msg }) = res {
+        assert_eq!(msg, "NOTUNIQUE Job not unique");
+    } else {
+        panic!("Expected protocol error.")
+    }
+
+    let had_job = consumer.run_one(0, &[queue_name]).unwrap();
+    assert!(had_job);
+    let had_another_one = consumer.run_one(0, &[queue_name]).unwrap();
+
+    // For the non-enterprise edition of Faktory, this assertion WILL FAIL:
```

Comment:
Won't it hit the `panic!` above first? Or, indeed, the `.unwrap_err`.

Reply:
Indeed,  there would not be any unique constraint violation, and asserting so would panic. Updated the comment to simply claim the queue is empty, showing the 'duplicate' job has not been queued indeed.